test/gtest/CMakeLists.txt

```diff
@@ -1,8 +1,10 @@
-find_package(GTest CONFIG REQUIRED)
+find_package(GTest CONFIG)
 if(GTest_FOUND)
   MESSAGE(STATUS "GoogleTest: " ${GTest})
-  enable_testing()
   include(GoogleTest)
+else()
+  message(WARNING "GoogleTest is not found! Skip GTests ... try install GTest or MIOpen dependencies.")
```

Comment:
I think we can introduce something like `MIOPEN_USE_GTEST`, make it enabled by default and fail if Gtest is not found. Then the user can explicitly disable or install GTest.

Reply:
`GTest`, like `rocTracer` are development tools instead of hard dependencies of the library. How about we deal with them in separate PR? I will adopt the other review options in this PR.

Reply:
If GTest is broken, then CI will silently omit many tests.

Reply:
I suggest we let the default `MIOpen` and `MIOpenDriver` target build without the GTest dependency, however, if tests are being built then GTest must be present otherwise compile will fail.

Reply:
+1